### PR TITLE
Add bazel CI workflow

### PR DESCRIPTION
**Pull request by mjcarroll**

### CODE REVIEW

**Comment by mjcarroll** [comment]
This is still expected to fail until I add bazel-specific tests.

File: .github/workflows/ci.bazelrc [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# This file contains Bazel settings to apply on CI only.
+# It is referenced with a --bazelrc option in the call to bazel in ci.yaml
```
Comment (rhaschke) [inline .github/workflows/ci.bazelrc:2]: I cannot find this reference. Can you please point me to the line in code?

**Review by rhaschke — COMMENTED** [review]
What's the purpose of the folder `test/bazel`? 
If that's just the build folder for bazel, I suggest to move it to `.github/workflows/bazel`.

**Comment by rhaschke** [comment]
Please integrate this PR into #350. They build upon each other anyway.

**Comment by mjcarroll** [comment]
> What's the purpose of the folder test/bazel?

It's where an integration test of the bazel rules will go, I haven't implemented it yet.   I will move it into `bazel/integration_test` as part of cleaning up the bazel-specific files.

File: .github/workflows/ci.bazelrc (hunk 1 above)
Comment (mjcarroll) [reply .github/workflows/ci.bazelrc:2]: Well, it used to be used as part of the upstream CI workflow, but when trying to find the reference, I find that it has been removed in https://github.com/bazel-contrib/.github/pull/11/files#diff-fcdc74e04f54e030b7b5cfde9bbcd1cc443162841efd9375481872e030b43a2bL153, which was reported upstream here: https://github.com/bazel-contrib/.github/issues/23)

**Pull request merged**